parsons/databases/database/database.py

```diff
@@ -0,0 +1,257 @@
+            return self.VARCHAR
+
+        # Exit early if it's None
+        # is_valid_sql_num(None) == False
+        # instead of defaulting to varchar (which is the next test)
+        # return the compare type
+        if val_lit is None:
+            return cmp_type
+
+        # Make sure that it is a valid integer
+        # Python accepts 100_000 as a valid form of 100000,
+        # however a sql engine may throw an error
+        if not self.is_valid_sql_num(value):
+            return self.VARCHAR
+
+        type_lit = type(val_lit)
+
+        # If a float, stop here
+        # float is highest after varchar
+        if type_lit == float or cmp_type == self.FLOAT:
+            return self.FLOAT
+
+        # The value is very likely an int
+        # let's get its size
+        # If the compare types are empty and use the types of the current value
+        if type_lit == int and cmp_type in (self.INT_TYPES + [None, ""]):
+
+            # Use smallest possible int type above TINYINT
+            if (self.SMALLINT_MIN < val_lit < self.SMALLINT_MAX):
+                return self.get_bigger_int(self.SMALLINT, cmp_type)
+            elif (self.MEDIUMINT_MIN < val_lit < self.MEDIUMINT_MAX):
+                return self.get_bigger_int(self.MEDIUMINT, cmp_type)
+            elif (self.INT_MIN < val_lit < self.INT_MAX):
+                return self.get_bigger_int(self.INT, cmp_type)
+            else:
+                return self.BIGINT
+
+        # Need to determine who makes it all the way down here
+        return cmp_type
+
+    def format_column(self, col, index=None, replace_chars=None,
+                      col_prefix=None):
+        """Format the column to meet database contraints.
+
+        Formats the columns as follows:
+            1. Coverts to lowercase (if case insensitive)
+            2. Strips leading and trailing whitespace
+            3. Replaces invalid characters
+            4. Renames if in reserved words
+        `Args`:
+            col: str
+                The column to format.
+            index: int
+                (Optional) The index of the column. Used if the column is empty.
+            replace_chars: dict
+                A dictionary of invalid characters and their replacements. If
+                ``None`` uses {" ": "_"}
+            col_prefix: str
+                The prefix to use when the column is empty or starts with an
+                invalid character.
+        `Returns`:
+            str
+                The formatted column.
+        """
+        REPLACE_CHARS = {" ": "_"}
+        replace_chars = replace_chars or REPLACE_CHARS
+
+        if not self.IS_CASE_SENSITIVE:
+            col = col.lower()
+
+        col = col.strip()
+
+        for char, rep_char in replace_chars.items():
+            col = col.replace(char, rep_char)
+
+        # The format for this column passes all other checks
+        if col == "":
+            return f"{col_prefix or '_'}{index or ''}"
```

Comment:
Shouldn't the default be `'col_'` rather than `_`?

Also, just curious when you're using the or pattern rather than argument defaults (e.g. `col_prefix='col_`)?

Reply:
This is in the base class. It's overridden in the subclasses to use `'col_'`. For example, in [Redshift](https://github.com/dannyboy15/parsons/blob/database-refactor/parsons/databases/redshift/rs_create_table.py#L173-L174).

> Also, just curious when you're using the or pattern rather than argument defaults (e.g. col_prefix='col_)?

I think this may be the only place where I did that. And you're right, I think it makes sense to throw it as an arg default.

Reply:
That would be great!